### ISSUE TITLE
Graph not displaying correct values. 

### ISSUE THREAD

**Issue by tomlut** [kalkih/mini-graph-card#117]
I'm seeing strange behaviour of the graph. It is reporting a high value when it shouldn't. See screen caps below:

![Annotation 2019-07-10 104738](https://user-images.githubusercontent.com/10679300/60932540-c5c7a100-a301-11e9-80f4-b88d0d3287a8.png)
![Annotation 2019-07-10 104822](https://user-images.githubusercontent.com/10679300/60932542-c5c7a100-a301-11e9-994e-f6c54b0d1256.png)

No amount of changing the points per hour seems to fix it.

```
      - color_thresholds:
          - color: '#0da035'
            value: -1
          - color: '#e0b400'
            value: 1800
          - color: '#e45e65'
            value: 2100
        color_thresholds_transition: hard
        entities:
          - sensor.lounge_dehumidifier_power
        group: false
        hour24: true
        line_width: 6
        name: Lounge Dehumidifier Power
        points_per_hour: 2
        show:
          fill: fade
          icon: true
          labels: false
          name: true
          state: true
        style: |
          ha-card {
            border: solid 2px var(--primary-color);
          }
        type: 'custom:mini-graph-card'
```

<s>EDIT: this appears to be a firefox issue. The graphs display correctly on Chrome and Safari. Clearing the cache on firefox (CTRL + F5) did not make any improvement.</s>

EDIT2: or maybe it is a caching issue. Opening the page in a private window in firefox does not have the problem.

**Comment by kalkih** [comment]
Could it be that the sensor has no history for that initial timespan? If there's no history data available at the beginning of the timespan we don't know how to fill the graph, and will give the "missing timespan" the value of the first known history entry, which in this case looks to be about 16? hours in.

The default history graph I believe does this differently and will fill the missing timespan with the value of the previous known history entry in existence, even if it's prior to the actual timeframe of the graph.

I would guess this is the reason to why you are seeing this behaviour, but could also be something else, hard to tell without the actual history data.

It's strange though, if it's behaving differently if cached or not, what did the graph look like when you opened it in incognito or another browser? We might have an inconsistency in how we compute the history data when it's cached versus fetched.

**Comment by tomlut** [comment]
My claim that Chrome was ok is false. It is happening there too now.

Looks like this in a previously opened browser:

![previously_open](https://user-images.githubusercontent.com/10679300/60968897-892b9200-a361-11e9-811d-8665fb91b28e.png)

Looks like this in a private tab:

![private](https://user-images.githubusercontent.com/10679300/60968923-95afea80-a361-11e9-9792-daf1cbee1d87.png)

The value is zero right to the beginning of the graph. Unchanged. So I guess there is no state there for the graph to read?

The built in history graph:

![builtin](https://user-images.githubusercontent.com/10679300/60969105-0eaf4200-a362-11e9-823a-8c004385817e.png)

**Comment by kalkih** [comment]
Okay thanks, that's interesting, so most likely an inconsistency in how we compute cached data vs fresh data.

**Comment by michaelblight** [comment]
I've seen this problem too. I created a version of the code where it optionally doesn't compress the storage to make it easier to view in Dev Tools. Here's an example of the problem (and yes, I do call my electric blanket 'blankie'):
<img width="465" alt="Screen Shot 2019-07-21 at 3 08 07 pm" src="https://user-images.githubusercontent.com/373079/61587234-9a7a6700-abc9-11e9-8ba6-5b6d309b4a56.png">
<img width="375" alt="Screen Shot 2019-07-21 at 3 07 51 pm" src="https://user-images.githubusercontent.com/373079/61587242-a9f9b000-abc9-11e9-9349-a709373efbdb.png">

The first load of this data was from:
`http://192.168.0.66:8123/api/history/period/2019-07-20T05:01:08.351Z?filter_entity_id=sensor.blankie_power&end_time=2019-07-21T05:01:08.351Z`
And it's first entry was at 2019-07-20T05:01:08.351000+00:00 with a state of 0. This is a 24-hour period, whereas I have `hours_to_show` set to 36. So this is possibly the problem.

Here's the local store showing 36 hours:
<img width="562" alt="Screen Shot 2019-07-21 at 3 07 16 pm" src="https://user-images.githubusercontent.com/373079/61587672-cef22100-abd1-11e9-8d0b-be80ee5b4c24.png">

I'm pretty sure I had it set to 24 hours before changing it to 36 and erasing the local store. I then increased it to 48, cleared the local store and refreshed and it looked like it only got 36 hours in the first call. I will do some more testing when my head doesn't hurt so much trying to convert AU time to UTC.

**Comment by Magalex2x14** [comment]
I see this problem too. Deleting local storage helps, but only until the first refresh of the page with the card. In addition, over time, the problematic graph is not updated with the card (night time does not shift to the left).
![Снимок экрана 2019-08-15 в 21 34 41](https://user-images.githubusercontent.com/4982025/63117871-551d3f80-bfa5-11e9-9dbf-08b9ce26be4a.png)

**Comment by kalkih** [comment]
This should now hopefully be fixed in the latest release.

**Comment by tomlut** [comment]
Sorry not fixed in 0.8.0 (browser cache reset).


![Screenshot_2019-10-21 Home Assistant(2)](https://user-images.githubusercontent.com/10679300/67168363-9429a000-f3ef-11e9-9c4f-962973071e59.png)
